calibr8/tests.py

```diff
@@ -689,37 +695,79 @@ def test_infer_independent(self):
-    def test_symbolic_loglikelihood(self):
+
+    @pytest.mark.skipif(not HAS_PYMC, reason='requires PyMC')
+    def test_symbolic_loglikelihood_checks_and_warnings(self):
         cmodel = _TestPolynomialModel(independent_key='S', dependent_key='A', mu_degree=1, scale_degree=1)
         cmodel.theta_fitted = [0, 1, 0.1, 1]
        
         # create test data
         x_true = numpy.array([1,2,3,4,5])
         y_obs = cmodel.predict_dependent(x_true)[0]
 
-        # create a pymc3 model using the calibration model
-        with pymc3.Model() as pmodel:
-            x_hat = pymc3.Uniform('x_hat', lower=0, upper=10, shape=x_true.shape, transform=None)
-
+        with pm.Model():
+            x_hat = pm.Uniform("x_hat", shape=5)
             with pytest.raises(ValueError, match="`name` must be specified"):
                 cmodel.loglikelihood(x=x_hat, y=y_obs)
 
             with pytest.warns(DeprecationWarning, match="Use `name` instead"):
-                L = cmodel.loglikelihood(x=x_hat, y=y_obs, replicate_id='A01', dependent_key='A')
-                assert isinstance(L, tt.TensorVariable)
+                cmodel.loglikelihood(x=x_hat, y=y_obs, replicate_id='A01', dependent_key='A')
+        pass
+
+    @pytest.mark.skipif(not HAS_PYMC, reason='requires PyMC')
+    def test_symbolic_loglikelihood(self):
+        cmodel = _TestPolynomialModel(independent_key='S', dependent_key='A', mu_degree=1, scale_degree=1)
+        cmodel.theta_fitted = [0, 1, 0.1, 1]
+       
+        # create test data
+        x_true = numpy.array([1,2,3,4,5])
+        y_obs = cmodel.predict_dependent(x_true)[0]
+
+        x_hat = at.vector()
+        x_hat.tag.test_value = x_true
+        L = cmodel.loglikelihood(x=x_hat, y=y_obs, name='L_A01_A')
+        assert isinstance(L, at.TensorVariable)
+        assert L.ndim == 0
+
+        # compare the two loglikelihood computation methods
+        x_test = numpy.random.normal(x_true, scale=0.1)
+        actual = L.eval({
+            x_hat: x_test
+        })
+        expected = cmodel.loglikelihood(x=x_test, y=y_obs)
+        assert numpy.ndim(expected) == 0
+        assert numpy.ndim(actual) == 0
+        numpy.testing.assert_almost_equal(actual, expected, 6)
+        pass
+
+    @pytest.mark.skipif(not HAS_PYMC, reason='requires PyMC')
+    def test_symbolic_loglikelihood_in_modelcontext(self):
+        cmodel = _TestPolynomialModel(independent_key='S', dependent_key='A', mu_degree=1, scale_degree=1)
+        cmodel.theta_fitted = [0, 0.5, 0.1, 1]
+       
+        # create test data
+        x_true = numpy.array([1,2,3,4,5])
+        y_obs = cmodel.predict_dependent(x_true)[0]
 
+        # create a PyMC model using the calibration model
+        with pm.Model() as pmodel:
+            x_hat = pm.Uniform("x_hat", 0, 1, shape=x_true.shape, transform=None)
             L = cmodel.loglikelihood(x=x_hat, y=y_obs, name='L_A01_A')
-            assert isinstance(L, tt.TensorVariable)
```

Comment:
Did I  misunderstand your previous explanation? I thought inside the model context, it's always a TensorVariable and not a distribution?

Reply:
In PyMC3 v3 the `pm.Normal()` and `pm.Normal.dist()` return different things.

But this test asserts _our_ `cmodel.loglikelihood()` to always return tensors.

Does that answer your question? Otherwise I might have misunderstood your question

Reply:
No, it's fine. And you dropped the test here because the assert is already placed elsewhere?

Reply:
Exactly